db/config.cc

```diff
@@ -970,6 +970,8 @@ db::config::config(std::shared_ptr<db::extensions> exts)
     , enable_repair_based_node_ops(this, "enable_repair_based_node_ops", liveness::LiveUpdate, value_status::Used, true, "Set true to use enable repair based node operations instead of streaming based.")
     , allowed_repair_based_node_ops(this, "allowed_repair_based_node_ops", liveness::LiveUpdate, value_status::Used, "replace,removenode,rebuild,bootstrap,decommission", "A comma separated list of node operations which are allowed to enable repair based node operations. The operations can be bootstrap, replace, removenode, decommission and rebuild.")
     , enable_compacting_data_for_streaming_and_repair(this, "enable_compacting_data_for_streaming_and_repair", liveness::LiveUpdate, value_status::Used, true, "Enable the compacting reader, which compacts the data for streaming and repair (load'n'stream included) before sending it to, or synchronizing it with peers. Can reduce the amount of data to be processed by removing dead data, but adds CPU overhead.")
+    , enable_tombstone_gc_for_streaming_and_repair_compaction(this, "enable_tombstone_gc_for_streaming_and_repair_compaction", liveness::LiveUpdate, value_status::Used, false,
+            "Enable compaction during streaming and repair (see enable_compacting_data_for_streaming_and_repair) to garbage-collect tombstones")
```

Comment:
The word "compaction" is misleading here. People will see it as a compaction process that reads some sstables and writes others. But here it's just about eliminating tombstones (and compacting_reader is just half of what users see as compaction.

Reply:
We need to mention that setting enable_tombstone_gc_for_streaming_and_repair_compaction alone is not enough. It works only when nable_compacting_data_for_streaming_and_repair is set to true.

I am wondering if it is better to combine both options. 

streaming_and_repair_compaction_mode = {
'disabled',
'data',
'data-and-tombstone'
}

Reply:
>  The word "compaction" is misleading here. People will see it as a compaction process that reads some sstables and writes others. But here it's just about eliminating tombstones (and compacting_reader is just half of what users see as compaction).

I can remove the compaction from the name.

> streaming_and_repair_compaction_mode = {
>'disabled',
>'data',
> 'data-and-tombstone'
>}

This flag is already in many releases, we cannot change its type now.

Reply:
It is sad but ok.